config/opal_check_pmi.m4

```diff
@@ -303,6 +303,19 @@ AC_DEFUN([OPAL_CHECK_PMIX],[
 
            # if it does exist, then we need to parse it to find
            # the actual release series
+           AS_IF([test "$opal_external_pmix_version_found" = "0"],
+                 [AC_MSG_CHECKING([version 4x])
+                  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+                                                      #include <pmix_version.h>
+                                                      #if (PMIX_VERSION_MAJOR != 4L)
+                                                      #error "not version 4"
+                                                      #endif
+                                                      ], [])],
+                                    [AC_MSG_RESULT([found])
+                                     opal_external_pmix_version=4x
+                                     opal_external_pmix_version_found=1],
+                                    [AC_MSG_RESULT([not found])])])
+
```

Comment:
Cool, this comes from [upstream here right](https://github.com/open-mpi/ompi/blob/master/config/opal_check_pmi.m4)?

I see there a lot more upstream changes too.  Do we want to bring those in as well?

Reply:
Yes, we can bring those in. But first I am trying to validate that with PMIx4, we can resolve the issues that we are seeing with PRRTE. I just checked this branch on Jenkins, it still fails but fails with a missing library (libcrypto) error.